### DISCUSSION TITLE
Push directive inside if not working in livewire v3

### DISCUSSION THREAD

**Discussion by RikoDEV** [livewire/livewire#6709] [7. Bugs (archived)] [answered]
### Livewire version

v3.0.2

### Which PHP version are you using?

PHP 8.2

### Repository URL

https://github.com/RikoDEV/livewire3-if-push

### Steps To Reproduce

While creating a form divided into steps, I encountered a problem with @push directive when moving to the next step. While the first `@if` performs `@push` correctly, the next ones compared to Livewire v2 do nothing. 

1. Run my example.
2. In first step, text `push 1 works` showing. Now click the button **One more step**.
3. `@push` directive inside if not working.

Layout:
```blade
<!DOCTYPE html>
<html>
    <head>
        <title>Dummy layout</title>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, user-scalable=yes, viewport-fit=cover">
        <meta name="csrf-token" content="{{ csrf_token() }}">

    </head>
    <body>
        {{ $slot}}

        @stack('scripts')
    </body>
</html>
```

Blade:
```blade
<div>
    @if ($step == 1)
        <div>
            <h1>Step 1!</h1>
            <p id="text1"></p>

            @push('scripts')
                <script>document.getElementById("text1").innerHTML = "push 1 works";</script>
            @endpush
        </div>
    @elseif ($step == 2)
       <div>
            <h1>Step 2!</h1>
            <p id="text2"></p>

            @push('scripts')
                <script>document.getElementById("text2").innerHTML = "push 2 works";</script>
            @endpush
        </div>
    @elseif ($step == 3)
        <div>
            <h1>Step 3!</h1>
            <p id="text3"></p>

            @push('scripts')
                <script>document.getElementById("text3").innerHTML = "push 3 works";</script>
            @endpush
        </div>
    @endif

    @if($step < 3)
        <button wire:click="addStep">One more step</button>
    @else
        <button disabled>No more, sorry</button>
    @endif
</div>
```

Component:
```php
<?php

namespace App\Livewire;

use Livewire\Component;

class FailingPushInsideIf extends Component
{
    public $step = 1;

    public function addStep()
    {
        $this->step = $this->step + 1;
    }

    public function render()
    {
        return view('livewire.failing-push-inside-if')->layout('layout');
    }
}
```

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by desgh2** [comment]
I have the same problem

**Comment by joshhanley** [comment]
@RikoDEV that's correct, Livewire can't dynamically load scripts, so you need to ensure they are already on the page during page load.

We previously have attempted a feature called dynamic stacks, which allowed us to push javascript to a stack on subsequent renders (like in your `@if` statement), but we could not get it to work nicely.

So for now, you're best to just include the `@push` on page load and not wrap it in an `@if`.

Hope this helps!

**Reply by RikoDEV** [reply]
Script was used here for an example, even if I put a div/img or other component in `@push` it will not be displayed. If this is the correct behavior of Livewire V3, it solves my question.
![nPHYm101tL](https://github.com/livewire/livewire/assets/18230443/550ff0c6-57da-4056-9087-4c1c7d7b0e01)

**Reply by joshhanley** [reply, marked as answer]
@RikoDEV Yeah that's correct, livewire can't dynamically load the stacks in the layout. Once the layout has been loaded, then it's set. Best to use `wire:navigate` to load your steps, in which it loads the layout each time.
